corehq/apps/users/validation.py

```diff
@@ -0,0 +1,52 @@
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+
+from corehq.apps.users.dbaccessors import user_exists
+from corehq.apps.users.exceptions import (
+    InvalidUsernameException,
+    ReservedUsernameException,
+    UsernameAlreadyExists,
+)
+from corehq.apps.users.util import format_username
+from corehq.apps.users.views.mobile import BAD_MOBILE_USERNAME_REGEX
+
+
+def validate_mobile_username(username, domain):
+    """
+    Returns the email formatted mobile username if valid
+    Otherwise raises custom exceptions based on the issue encountered with the username
+    :param username: str, required
+    :param domain: str, required
+    :return: str, email formatted mobile username
+    Example: validate_mobile_username('username', 'domain') -> 'username@domain.commcarehq.org'
+    """
+    if username is None:
+        raise ValueError
```

Comment:
This is a generic error that could be raised in other places for other reasons. What do you think of using a more specific error type to indicate that a username is required?

Reply:
This was my idea to use value error but in hindsight probably better to use a custom exception. Sorry @gherceg

Reply:
No worries! I just reused the InvalidUsernameException.